### ISSUE TITLE
Tracking Issue `--extern` crate modifiers

### ISSUE THREAD

**Issue by petrochenkov** [rust-lang/rust#98405]
This is a tracking issue for command line option syntax currently used by several modifiers affecting behavior of extern crates.
See [priv](https://github.com/rust-lang/rust/issues/98399), [noprelude](https://github.com/rust-lang/rust/issues/98398) and [nounused](https://github.com/rust-lang/rust/issues/98400) for their specific tracking issues.

The currently used syntax is
```
--extern [MODIFIERS:]NAME[=PATH]
```

Before stabilization it should likely be harmonized with command line syntax already used for native library modifiers, see item (3) in https://github.com/rust-lang/rust/pull/93901#issuecomment-1041325522.
- Modifiers can be converted to `+/-` form
  - `priv` - `+priv` or `-pub`
  - `noprelude` - `-prelude`
  - `nounused` - `+used`
- Position of the `MODIFIERS` component in the option can also be changed.

**Comment by jsgf** [comment]
> Before stabilization it should likely be harmonized with command line syntax already used for native library modifiers, see item (3) in https://github.com/rust-lang/rust/pull/93901#issuecomment-1041325522.
> Modifiers can be converted to +/- form 

I don't really see the relationship between `#[link]` and `--extern flag:`. The latter controls how a Rust dependency is handled/consumed by this particular rustc invocation, whereas the former controls how an external non-Rust dependency is finally linked.

(I also think the whole `#[link]` mechanism is a bad idea and should be deprecated, not extended. Controlling how things are ultimately linked is the responsibility of the build system, and so the functionality should be moved to Cargo.)

**Comment by ehuss** [comment]
I don't think that is referring to `#[link]`, just that it should use the same `+`/`-` syntax of the `-l` flag, like `--extern +prev,-prelude:foo=libfoo.rlib` or possibly rearranged to `--extern foo:+prev,-prelude=libfoo.rlib`
